services/src/api/handlers/layers.rs

```diff
@@ -314,6 +334,218 @@ async fn list_collection_handler<C: ApplicationContext>(
     Ok(web::Json(collection))
 }
 
+// Returns the search capabilities of the given provider
+#[utoipa::path(
+    tag = "Layers",
+    get,
+    path = "/layers/collections/search/capabilities/{provider}",
+    responses(
+        (status = 200, description = "OK", body = SearchCapabilities,
+            example = json!({
+                "search_types": {
+                    "fulltext": true,
+                    "prefix": true
+                },
+                "autocomplete": true,
+                "filters": [],
+            })
+        )
+    ),
+    params(
+        ("provider" = DataProviderId, description = "Data provider id")
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+async fn search_capabilities_handler<C: ApplicationContext>(
+    app_ctx: web::Data<C>,
+    path: web::Path<DataProviderId>,
+    session: C::Session,
+) -> Result<impl Responder> {
+    let provider = path.into_inner();
+
+    if provider == ROOT_PROVIDER_ID {
+        return Err(NotImplemented {
+            message: "Global search is not supported".to_string(),
```

Comment:
wouldn't users expect this to work?

Reply:
We should think of this separately. How would you search if you don't know the capabilities of the providers? Search only in the ones that have a search? Which type to provide?